util/election.go

```diff
@@ -0,0 +1,112 @@
+package util
+
+import (
+	"fmt"
+	"github.com/timescale/prometheus-postgresql-adapter/log"
+	"net/http"
+	"sync"
+)
+
+// Election defines an interface for adapter leader election.
+// If you are running Prometheus in HA mode where each Prometheus instance sends data to corresponding adapter you probably
+// want to allow writes into the database from only one adapter at the time. We need to elect a leader who can write to
+// the database. If leader goes down, another leader is elected. Look at `lock.go` for an implementation based on PostgreSQL
+// advisory locks. Should be easy to plug in different leader election implementations.
+type Election interface {
+	Id() string
+	BecomeLeader() (bool, error)
+	IsLeader() (bool, error)
+	Resign() error
+}
+
+// RestElection is a REST interface allowing to plug in any external leader election mechanism.
+// Remote service can use REST endpoints to manage leader election thus block or allow writes.
+type RestElection struct {
+	leader bool
+	mutex  sync.RWMutex
+}
+
+func NewRestElection() *RestElection {
+	r := &RestElection{}
+	http.Handle("/admin/election/leader", r.handleLeader())
+	http.Handle("/admin/election/resign", r.handleResign())
+	return r
+}
+
+func (r *RestElection) handleLeader() http.HandlerFunc {
+	return func(response http.ResponseWriter, request *http.Request) {
+		switch request.Method {
+		case http.MethodGet:
+			leader, err := r.IsLeader()
+			if err != nil {
+				log.Error("msg", "Failed on leader check", "err", err)
+				http.Error(response, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			fmt.Fprintf(response, "%v", leader)
+		case http.MethodPost:
```

Comment:
Seems a bit weird to POST something without content. Would have expected a PUT (POST typically to create a resource) where the content is the same type as what you return in the GET.

So, you, e.g., PUT a value of `1` to `/admin/election/leader` to become leader and a `0` to resign. A GET simply returns the current value. No resign path needed.

Otherwise it is not really a "RESTful" API.